### ISSUE TITLE
Extend `AutoCorrect: contextual` to be aware of non-LSP editor integrations

### ISSUE THREAD

**Issue by maxjacobson** [rubocop/rubocop#14539]
## Is your feature request related to a problem? Please describe.

https://github.com/rubocop/rubocop/pull/12657 introduced the fantastic idea of contextual autocorrect. When RuboCop knows it is running in an editor, there are some autocorrects which are a bit annoying, and it wisely chooses not to run them.

Personally, I haven't yet made the leap to using RuboCop via an LSP. Maybe I should. For now, I'm still using [ale](https://github.com/dense-analysis/ale), the "Asynchronous Lint Engine" for vim.

Here's a sample configuration:

```vim
" vim-plug setup
" https://github.com/junegunn/vim-plug
call plug#begin()
Plug 'dense-analysis/ale'
call plug#end()

" only run the linters I specify (do not infer anything)
let g:ale_linters_explicit = 1

" run rubocop when the filetype is ruby
let g:ale_linters= {'ruby': ['rubocop']}

" run rubocop's autocorrect on ruby files
let g:ale_fixers = {'ruby': ['rubocop']}

" opt in to autocorrect on save
let g:ale_fix_on_save = 1
```

It will run something like ([source](https://github.com/dense-analysis/ale/blob/4217461c4859c4fac31ce3d2e22574218f244cf5/autoload/ale/fixers/rubocop.vim#L22-L38)):

```
rubocop --auto-correct --force-exclusion --stdin
```

This is not LSP mode, and so the cops which are configured with `AutoCorrect: contextual` will run. This can be annoying while writing code like:

```ruby
def call
  results = some_process
end
```

When I save, this gets autocorrected to:

```ruby
def call
  some_process
end
```

This is sometimes a bit frustrating, because I'm like "hey wait where did that variable go?? I was about to use it!"

## Describe the solution you'd like

I'd like for `AutoCorrect: contextual` to behave the same when running RuboCop via an LSP or running RuboCop via ale.

Some potential solutions to consider:

1. When the `--stdin` flag is enabled, is that a signal that the user is likely using an editor integration? If so, when that flag is passed, contextually-auto-correctable cops should not autocorrect
2. Alternatively, rubocop could introduce a new flag which is something like `--auto-correct-context editor` and when that flag is passed, contextually-auto-correctable cops should not autocorrect. If such a flag existed, editor integrations could use it

## Describe alternatives you've considered

1. switch to using an LSP
2. in my projects, configure these cops as `AutoCorrect: false` so they do not auto-correct at all (this would be a bit sad, because then the pre-commit hook also does not auto-correct them)

**Comment by koic** [comment]
For that kind of use case, the `--editor-mode` command line option was introduced in https://github.com/rubocop/rubocop/pull/12682. Please try adding it as shown below:

```console
rubocop --auto-correct --editor-mode --force-exclusion --stdin
```

https://docs.rubocop.org/rubocop/1.80/configuration.html#contextual

**Comment by maxjacobson** [comment]
ah! my bad, I totally missed that. I'll take a look now.

**Comment by maxjacobson** [comment]
Fabulous. I can resolve my issue by simply adding this line of ale vim config:

```vim
let g:ale_ruby_rubocop_options = "--editor-mode"
```

Perhaps that should be the default behavior for ale. It seems like that flag has existed for long enough that hopefully it wouldn't be a breaking change for them to enable it by default 🤔

I think the reason I missed this is that I was reading the code, and grepped for contextual, and found this: https://github.com/rubocop/rubocop/blob/2f5e01f379702b7d4fb4219b8e5a1b7a3a6af465/lib/rubocop/cop/autocorrect_logic.rb#L38-L40

If that comment mentioned `--editor-mode`, I would have discovered it that way. 

Regardles... thank you! I'll close this out.

**Comment by koic** [comment]
Ah, adjusting the comment sounds good. Anyway, I’m relieved the issue is resolved. Enjoy hacking!
